core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoinExec.scala

```diff
@@ -183,17 +181,45 @@ case class HashJoinExec(leftKeys: Seq[Expression],
   // return empty here as code of required variables is explicitly instantiated
   override def usedInputs: AttributeSet = AttributeSet.empty
 
+  private def findShuffleDeps(buildRDDs: Seq[RDD[_]]) : Seq[Dependency[_]] = {
+    val buildShuffleDeps: Seq[Dependency[_]] = buildRDDs.flatMap(_.dependencies
+      .filter(_.isInstanceOf[ShuffleDependency[_, _, _]]))
+    val oneToOneRdds = buildRDDs.flatMap(_.dependencies).
+      filter(_.isInstanceOf[OneToOneDependency[_]]).map(_.rdd)
+    if (oneToOneRdds.isEmpty) {
+      buildShuffleDeps
+    } else {
+      buildShuffleDeps ++ findShuffleDeps(oneToOneRdds)
+    }
+  }
```

Comment:
Confused about 1. Why should there be a shuffle on a replicated table? Shouldn't the fix be that the partitioning of that should be such that it is compatible with other kinds of partitioning? The only exception could be RangePartitioning. Is RangePartitioning the problem here? If its a regular ClusterDistribution then the fix should be to ensure that replicated table is not shuffled.

About 2, if the problem is wrapping inside another RDD, the code should look like in SnappySession.findShuffleDependencies that doesn't do any specific checks like here for OneToOneDependency which will likely fail in cases where a PruneDependency or RangeDependency gets created for example.